_posts/2020-09-21-dsc-release.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: post
+title: "DSC + Puppet: Arrived!"
+author: michaeltlombardi
+categories:
+  - news
+  - roadmap
+tags:
+  - powershell
+  - dsc
+  - windows
+  - forge
+---
+
+Earlier this year, we [announced an upcoming project][initial-blog] to drastically improve the story for using DSC Resources with Puppet.
+Today, we're announcing the general availability of that project!
+
+The [`Puppet.Dsc` PowerShell module][dsc-repo] is now available on the [PowerShell Gallery][dsc-gallery];
+furthermore, an initial batch of **twenty-six** PowerShell modules with DSC Resources have been Puppetized and [published to the Forge][dsc-forge]!
+
+We picked several of the most-downloaded modules with DSC Resources released in the last year to start with, but in the coming weeks you should find _everything_ available on the PowerShell Gallery up on the Forge!
+
+But today, in addition to sharing this news, I want to go into a bit more detail, both with what's going on under the hood and with how you can use these auto-generated modules.
+
+### Installing the Modules
+
+If you just want to get your hands on the PowerShell module to read the help and experiment on your own, you can install it like any other PowerShell module:
+
+```powershell
+Install-Module -Name Puppet.Dsc
+Import-Module -Name Puppet.Dsc
+Get-Command -Module Puppet.Dsc
+Get-Help -Name New-PuppetDscModule -ShowWindow
+```
+
+Some of the functions from the PowerShell module rely on the [Puppet Development Kit (PDK)][pdk-docs] being installed;
+we suggest using [Chocolatey][choco-site] to do so, but you can also [manually download and install][pdk-download] if you prefer.
+
+```bash
+choco install pdk
+```
+
+If you want to do the same sort of thing with one of the Puppetized modules from the forge, you can install them and treat them just like any other Puppet module:
+
+```bash
+# For example, installing the Puppetized ComputerManagementDsc module manually:
+puppet module install dsc-computermanagementdsc --Version 8.4.0-0-0
+# Or via r10k / Code Manager / Bolt:
+mod 'dsc-computermanagementdsc', '8.4.0-0-0'
+```
+
+## Puppetizing a PowerShell module
+
+So, you want to turn a PowerShell module with DSC Resources into a Puppet module?
+You can do this with a single command:
+
+```powershell
+# If you run this command without the PassThru flaf it returns nothing to the output stream.
```

Comment:
```suggestion
# If you run this command without the PassThru flag it returns nothing to the output stream unless there is an error during conversion.
```

Reply:
Fixed `flag` but PowerShell actually sends errors (and verbose, warning, etc each) to their own stream separate from output!